main.tf

```diff
@@ -41,8 +87,15 @@ module "cos" {
   cos_plan          = var.cos_plan
 }
 
+data "ibm_resource_instance" "existing_studio_instance" {
+  provider   = ibm.deployer
+  count      = var.existing_studio_instance != null ? 1 : 0
+  identifier = var.existing_studio_instance
+}
+
 resource "ibm_resource_instance" "studio_instance" {
   provider          = ibm.deployer
+  count             = var.existing_studio_instance != null ? 0 : 1
```

Comment:
Could you add `moved { .... }` block to avoid destructive behavior on an upgrade?

Reply:
I didn't rename any resource, what do you mean with `destructive behavior on an upgrade`? According to https://developer.hashicorp.com/terraform/language/moved:

```
The moved block programmatically changes the address of a resource
```

I don't think this is the case.

Reply:
The address changes from `ibm_resource_instance.studio_instance` to  `ibm_resource_instance.studio_instance[0]`. That will trigger destroy/create.

Reply:
See https://developer.hashicorp.com/terraform/language/modules/develop/refactoring#enabling-count-or-for_each-for-a-resource